src/chunkedstorage.cc

```diff
@@ -64,21 +64,23 @@ void Writer::addToBlock( void const * data, size_t size )
 
 void Writer::saveCurrentChunk()
 {
-  size_t maxCompressedSize = compressBound( bufferUsed );
+  size_t maxCompressedSize = ZSTD_compressBound( bufferUsed );
 
   if ( bufferCompressed.size() < maxCompressedSize )
     bufferCompressed.resize( maxCompressedSize );
 
-  unsigned long compressedSize = bufferCompressed.size();
 
-  if ( compress( &bufferCompressed.front(), &compressedSize, &buffer.front(), bufferUsed ) != Z_OK )
+  const size_t size_or_err =
```

Comment:
API design is different.

In zlib,

compress will write the size written to its 2nd paramater

In Zstd,

compress will return the size written or error code. https://github.com/facebook/zstd/issues/1825#issuecomment-542921358